test1474: disable test on NetBSD, OpenBSD and Solaris 10

These kernels only send a fraction of the requested amount of the first
large block, invalidating the assumptions of the test and causing it to
fail.

https://github.com/curl/curl/commit/08f9b21483b67a1180f342eec4441a34f04745a5
... and related commits

Index: tests/data/test1474
--- tests/data/test1474.orig
+++ tests/data/test1474
@@ -23,13 +23,20 @@
 # builds which are often run on overloaded servers.
 # Increasing the --limit-rate would decrease the test time, but at the cost of
 # becoming even more sensitive to delays (going from 500 msec to 250 msec or
-# less of accepted delay before failure).
+# less of accepted delay before failure).  Adding a --speed-time would increase
+# the 1 second delay between writes to longer, but it would also increase the
+# total time needed by the test, which is already quite high.
+#
+# The assumption in step 3 is also broken on NetBSD 9.3, OpenBSD 7.3 and
+# Solaris 10 as they only usually send about half the requested amount of data
+# (see https://curl.se/mail/lib-2023-09/0021.html).
 <info>
 <keywords>
 HTTP
 HTTP PUT
 Expect
 flaky
+timing-dependent
 </keywords>
 </info>
 # Server-side
@@ -81,6 +88,9 @@ HTTP PUT with Expect: 100-continue and 417 response du
  <command>
 http://%HOSTIP:%HTTPPORT/we/want/%TESTNUMBER -T %LOGDIR/test%TESTNUMBER.txt --limit-rate 64K --expect100-timeout 0.001
 </command>
+<precheck>
+perl -e "print 'Test does not work on this BSD system' if ( $^O eq 'netbsd' || $^O eq 'openbsd' || ($^O eq 'solaris' && qx/uname -r/ * 100 <= 510));"
+</precheck>
 # Must be large enough to trigger curl's automatic 100-continue behaviour
 <file name="%LOGDIR/test%TESTNUMBER.txt">
 %repeat[132 x S]%%repeat[16462 x xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx%0a]%
